app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  def authenticate_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
Thanks. I am going to implement that today. Have to read up on this topic first though.